src/btree/bt_sync.c

```diff
@@ -285,6 +316,12 @@ __wt_sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
             if (walk == NULL)
                 break;
 
+            /* Traverse through the internal page for obsolete child pages. */
+            if (is_las && WT_PAGE_IS_INTERNAL(walk->page)) {
+                WT_WITH_PAGE_INDEX(session, ret = __sync_ref_int_obsolete_cleanup(session, walk));
+                WT_ERR(ret);
+            }
+
```

Comment:
The chunk of code farther down:
```
/*  
 * Check whether the tree is lookaside btree and verify whether the page contents
 * are obsolete or not by checking the visibility of the page stop time pair.
 */  
if (is_las && __sync_ref_is_obsolete(session, walk)) {
    /*  
     * The duplicate tree walk code expects the ref state to be in memory. We need
     * to get the duplicate tree walk pointer before marking the current tree page
     * as deleted.
     */  
    WT_ERR(__sync_dup_walk(session, walk, flags, &prev));

    /* Mark the page as deleted */
    WT_ERR(__sync_ref_mark_deleted(session, walk));
                                                
    /*  
     * The duplicate tree walk pointer, prev, was obtained before we marked the page
     * as deleted. Continue the rest of the tree walk using that pointer.
     */  
    goto skipwalk;
}   
continue;
```
Can this still happen?

If it's an internal page, we can't mark it deleted, if it's a leaf page, by definition it has to be in memory, otherwise walk wouldn't have returned it to us?

Reply:
No, this code is not required now. But this code comes back again when we are handling the in-memory pages that are needed to be added to urgent eviction queue.